src/app/core/store/documents/documents.action.ts

```diff
@@ -110,19 +110,19 @@ export namespace DocumentsAction {
   export class UpdateFailure implements Action {
     public readonly type = DocumentsActionType.UPDATE_FAILURE;
 
-    public constructor(public payload: {error: any}) {}
+    public constructor(public payload: {error: any; originalDocument?: DocumentModel}) {}
   }
 
   export class UpdateData implements Action {
     public readonly type = DocumentsActionType.UPDATE_DATA;
 
-    public constructor(public payload: {document: DocumentModel}) {}
+    public constructor(public payload: {document: DocumentModel; originalDocument?: DocumentModel}) {}
```

Comment:
You do not need to pass original document here. It would be better to get its latest version in the effect pipeline and then put it directly to the failure action.